### ISSUE TITLE
error when pgn has two annotations in a row

### ISSUE THREAD

**Issue by trevor-ofarrell** [mliebelt/PgnViewerJS#195]
`Expected " ", ".", "O-O", "O-O-O", "x", "{", [RNBQKP], or [a-h] but "-" found.
Could not find startPlay: 1` this error is triggered by the following PGN:

> PGN: [Event "Rated Blitz game"] [Site "https://lichess.org/UM4fJExy"] [Date "2021.03.12"] [White "Schuhmacherr"] [Black "drainer9"] [Result "0-1"] [UTCDate "2021.03.12"] [UTCTime "00:02:12"] [WhiteElo "1809"] [BlackElo "1803"] [WhiteRatingDiff "-6"] [BlackRatingDiff "+6"] [Variant "Standard"] [TimeControl "180+0"] [ECO "B03"] [Opening "Alekhine Defense"] [Termination "Normal"] [Annotator "lichess.org"] 1. e4 { [%clk 0:03:00] } 1... Nf6 { [%clk 0:03:00] } 2. e5 { [%clk 0:02:59] } 2... Nd5 { [%clk 0:03:00] } 3. d4 { [%clk 0:02:58] } 3... d6 { [%clk 0:02:59] } { B03 Alekhine Defense } 4. f4 { [%clk 0:02:56] } 4... dxe5 { [%clk 0:02:59] } 5. fxe5 { [%clk 0:02:55] } 5... Bf5 { [%clk 0:02:58] } 6. Nf3 { [%clk 0:02:54] } 6... Nc6 { [%clk 0:02:58] } 7. Bc4 { [%clk 0:02:52] } 7... e6 { [%clk 0:02:57] } 8. O-O { [%clk 0:02:49] } 8... Ncb4 { [%clk 0:02:54] } 9. Bg5 { [%clk 0:02:45] } 9... f6 { [%clk 0:02:52] } 10. exf6 { [%clk 0:02:44] } 10... gxf6 { [%clk 0:02:52] } 11. Bh4 { [%clk 0:02:37] } 11... Nxc2 { [%clk 0:02:50] } 12. Re1 { [%clk 0:02:24] } 12... Nxa1 { [%clk 0:02:48] } 13. Nc3 { [%clk 0:02:20] } 13... Nxc3 { [%clk 0:02:26] } 14. Qxa1 { [%clk 0:02:12] } 14... Nd5 { [%clk 0:02:24] } 15. Qd1 { [%clk 0:01:52] } 15... Be7 { [%clk 0:02:12] } 16. Nd2 { [%clk 0:01:39] } 16... Qd7 { [%clk 0:01:56] } 17. Qh5+ { [%clk 0:01:37] } 17... Bg6 { [%clk 0:01:53] } 18. Qg4 { [%clk 0:01:28] } 18... Bf5 { [%clk 0:01:45] } 19. Qh5+ { [%clk 0:01:26] } 19... Bg6 { [%clk 0:01:42] } 20. Qg4 { [%clk 0:01:25] } 20... Bf7 { [%clk 0:01:34] } 21. Qg7 { [%clk 0:00:45] } 21... Rg8 { [%clk 0:01:26] } 22. Qxh7 { [%clk 0:00:44] } 22... O-O-O { [%clk 0:01:25] } 23. Ne4 { [%clk 0:00:31] } 23... Bg6 { [%clk 0:01:21] } 24. Qh6 { [%clk 0:00:22] } 24... Bxe4 { [%clk 0:01:16] } 25. Rxe4 { [%clk 0:00:19] } 25... Bd6 { [%clk 0:01:06] } 26. Qxf6 { [%clk 0:00:17] } 26... Rde8 { [%clk 0:01:00] } 27. Qf3 { [%clk 0:00:13] } 27... Nb6 { [%clk 0:00:56] } 28. Bb3 { [%clk 0:00:11] } 28... Qb5 { [%clk 0:00:49] } 29. Rxe6 { [%clk 0:00:06] } 29... Ref8 { [%clk 0:00:42] } 30. Qh3 { [%clk 0:00:04] } 30... Rf1# { [%clk 0:00:41] } { Black wins by checkmate. } 0-1

After removing { Black wins by checkmate. } the code works as expected. 
![image](https://user-images.githubusercontent.com/47125174/111689455-47ffac80-87e9-11eb-9047-4df2677d4cb4.png)

**Comment by mliebelt** [comment]
Thank you for creating that ticket. I have checked  it with the current state of the grammar (see [test run including test case line 112](https://github.com/mliebelt/pgn-parser/runs/2146061998?check_suite_focus=true)). I have added there an additional test case that has your PGN (without tags), and that can be read by the grammar.

The reason for that is that PgnViewerJS does not use the current version of pgn-parser. I have invested some time in the grammar around comments, and your case seems to be one of the faulty ones. I will check how to update the dependency to pgn-parser, without too much invest.
